### PR TITLE
Fix images not saved for enquiry

### PR DESCRIPTION
**Pull request by StupidCodeFactory**
## Description

The `DateConcern` was causing the `Enquiry` model all contain changes. When attaching an document Rails would never call `save` on the object detecting that it had other changes and hence on changing the attributes in memory.

```ruby
# this initialises other :date_received_day, :date_received_month, :date_recieved_year
# on on every object instantiation causing enquiry#changes? to be true  
date_attributes :date_received 
```

I've removed the `DateConcern` from the `Enquiry` and fixed the Javascript that set date to today or tomorrow. 

Fixed the "Other received type" which was not accessible by adding a label text and applying the "govuk-visually-hidden" class.
This field was also not retained when the form re-displays for validation errors.

## Notes 
It was impossible to write an integration test to reproduce the bug as it would save the attachement correctly 🤷‍♂️

### CODE REVIEW

**Review by slorek — APPROVED** [review]

**Review by nasirkhanbeis — APPROVED** [review]
looks good as expected

File: config/locales/en.yml [hunk 1]
```diff
@@ -350,6 +350,14 @@ en:
       collaboration/access/read_only: "readonly"
     errors:
       models:
+        investigation/enquiry:
+          attributes:
+            date_received:
+              blank: Enter the date the enquiry was received
+              must_be_real: Date received of enquiry must be a real date
+              incomplete: Date received of enquiry must include a %{missing_date_parts}
+              in_future: Date received of enquiry must be today or in the past
```
Comment (frankieroberto) [inline config/locales/en.yml:359]: Think these need a bit of re-wording...

Suggestion:

```suggestion
              must_be_real: Date enquiry received must be a real date
              incomplete: Date enquiry received must include a %{missing_date_parts}
              in_future: Date enquiry received must be today or in the past
```

**Review by frankieroberto — CHANGES_REQUESTED** [review]
Looks good, thanks!

Just needs some tweaks to the error messages, I think.

File: config/locales/en.yml (hunk 1 above)
Comment (StupidCodeFactory) [reply config/locales/en.yml:359]: cheers! yes does read much better indeed.

**Review by frankieroberto — APPROVED** [review]

**Pull request merged**